### PR TITLE
deps(nuxt): migrate to nuxt 4.5.1 and vite 8 (rolldown)

### PR DESCRIPTION
**Pull request by DysektAI**
## Summary

Migrates the toolchain to **Nuxt 4.5.1 on Vite 8 (Rolldown-powered)** — the dependency update that dependabot PR #629 has been blocked on (`@nuxt/vite-builder@4.5.1` requires `vite ^8.1.5` and imports `transformWithOxc`, but the repo's `vite: '^7.3.5'` override in `pnpm-workspace.yaml` was forcing `vite@7.3.6`, causing the CI failure).

**Vite 7 → 8 breaking changes addressed** (per the [official migration guide](https://vite.dev/guide/migration)):
- `build.rollupOptions.output.manualChunks` (function form, deprecated in Vite 8) → `build.rolldownOptions.output.codeSplitting.groups` — the three vendor groups (leaflet / supabase / core) are now regex `test` entries. Verified parity: leaflet lands in a dedicated ~148 KB chunk, same as before.

## Changes

- **`pnpm-workspace.yaml`**: `vite` override `^7.3.5` → `^8.1.5`; `vue-i18n` catalog `^11.3.2` → `^11.4.8`
- **`package.json` / lockfile**: `nuxt` `^4.5.1`, `@nuxt/ui` `^4.10.0`, `@nuxtjs/i18n` `^10.6.0`, `@nuxtjs/sitemap` `^8.3.2`, `vue-router` `^5.2.0`, `@nuxt/test-utils` `^4.1.0` (nuxt 4.5 support). All ecosystem peers now declare `vite ^8` support (`@vitejs/plugin-vue@6.0.8`, `@tailwindcss/vite@4.3.3`, `vitest` `^6 || ^7 || ^8`).
- **`nuxt.config.ts`**: `manualChunks` → `rolldownOptions.output.codeSplitting.groups`
- **`tests/test-setup.ts`**: Nuxt 4.5 changed `$fetch` from a global to an auto-imported module binding captured at import time (`export const $fetch = globalThis.$fetch`), so per-test `vi.stubGlobal('$fetch', …)` no longer intercepted it (6 store test files broke). The setup now installs a call-time delegator that forwards to the current global (per-test stub) or falls back to the URL-router mock.

## Validation (all local)

- `pnpm run build` (production, CI-mode env) — **passes on vite 8.2.0 / rolldown**
- `pnpm run test` — **227 files / 2221 tests pass**
- `pnpm run test:api-gateway` — 141 pass
- `pnpm run typecheck`, `pnpm run lint`, `i18n:check`, `fallow audit` (0 introduced), `lint-blank-lines` — all clean
- Vendor chunking parity verified (leaflet chunk ≈ 148 KB, same as vite 7 build)
- `@nuxt/ui@4.10.0` still ships `useResizable.js` (the existing shim plugin target) — no shim change needed
- Node 24.12.0 satisfies nuxt 4.5.1 engine range (`^22.18 || ^24.11 || >=26`)

## Notes

- Unblocks dependabot PR #629 (nuxt-ecosystem group).
- Known watch-items from the Vite 8 guide: CJS default-import interop changes (escape hatch `legacy.inconsistentCjsInterop: true` if any edge case surfaces), Oxc minifier / LightningCSS output differences — all covered by the build + test + Lighthouse + bundle-analysis CI gates.
- Not merged until CI is fully green and reviews are clean.

<a href="https://cubic.dev/pr/tarkovtracker-org/TarkovTracker/pull/638?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Performance**
  * Improved application build organization to support more efficient loading of map, authentication, and Vue-related functionality.

* **Compatibility**
  * Updated the Nuxt, Vue, routing, internationalization, UI, and sitemap components to newer supported versions.

* **Reliability**
  * Improved request handling in automated testing, helping ensure more consistent validation of application behavior.

<details open><summary><h3>Greptile Summary</h3></summary>

This PR migrates the project from Nuxt 4.4/Vite 7 to Nuxt 4.5.1/Vite 8 (Rolldown), unblocking the dependabot ecosystem group. It also fixes a test regression introduced by Nuxt 4.5's change to `$fetch` from a mutable global to a module-captured binding.

- **`pnpm-workspace.yaml`**: Lifts the `vite` override to `^8.1.5` and drops the now-resolved `@nuxt/test-utils>h3-next` patch; `vue-i18n` catalog bumped to `^11.4.8`.
- **`nuxt.config.ts`**: Replaces the deprecated `build.rollupOptions.output.manualChunks` function with Rolldown's `build.rolldownOptions.output.codeSplitting.groups` declarative API; three vendor groups (leaflet, supabase, core) are preserved.
- **`tests/test-setup.ts`**: Introduces a call-time delegator so tests using `vi.stubGlobal('$fetch', …)` still intercept calls even though Nuxt 4.5 captures the binding at module import time.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; the migration is mechanically correct, all tests pass, and no breaking changes are introduced.

All three substantive changes (vite override, rolldownOptions chunk groups, $fetch delegator) are correctly implemented. The build and full test suite were validated locally. The only comment left is a note about the vendor-core regex scope narrowing compared to the old manualChunks logic — it could affect cache granularity for packages like vue-router but does not break anything.

**Files Needing Attention:** nuxt.config.ts — the vendor-core codeSplitting regex is narrower than the old manualChunks check and may leave packages like vue-router out of the named chunk.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| nuxt.config.ts | Migrates chunk splitting from rollupOptions.manualChunks to rolldownOptions.output.codeSplitting.groups; the new regex for vendor-core requires a trailing slash after the module name, so packages like vue-router and pinia-plugin-persistedstate are no longer explicitly grouped there. |
| tests/test-setup.ts | Replaces direct vi.stubGlobal('$fetch', mockFetch) with a call-time delegator; correctly handles Nuxt 4.5's change of $fetch from a mutable global to a module-captured binding. |
| pnpm-workspace.yaml | Bumps vite override from ^7.3.5 to ^8.1.5, updates vue-i18n catalog to ^11.4.8, and removes the now-unnecessary @nuxt/test-utils>h3-next override. |
| package.json | Bumps nuxt to ^4.5.1 and associated ecosystem packages (@nuxt/ui, @nuxtjs/i18n, @nuxtjs/sitemap, vue-router, @nuxt/test-utils) to versions that declare vite ^8 compatibility. |
| .nuxtrc | Updates test-utils setup version pin from 4.0.3 to 4.1.0 to match the package.json devDependency bump. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["Test code calls $fetch(url, init)"] --> B["Module-captured binding\n(set at import time → delegator)"]
    B --> C["delegator() runs"]
    C --> D{"globalThis.$fetch\n=== delegator?"}
    D -- "Yes (no per-test stub)" --> E["call mockFetch(url, init)\n(URL-router mock)"]
    D -- "No" --> F{"vi.isMockFunction\n(globalThis.$fetch)?"}
    F -- "No (real function or undefined)" --> E
    F -- "Yes" --> G["Forward to per-test\nvi.stubGlobal mock"]
    G --> H["Per-test mock returns"]
    E --> I["mockFetch returns\nURL-matched response"]
```
</details>

<sub>Reviews (4): Last reviewed commit: ["chore(deps): drop stale h3-next override..."](https://github.com/tarkovtracker-org/tarkovtracker/commit/20a1099ae7e7c08ba5b63704fb0054170ce3b3a5) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49695272)</sub>

### CODE REVIEW

**Comment by DysektAI** [comment]
Addressed in d0ca4474 — removed the explanatory comment block from `tests/test-setup.ts` per the repo's no-comments guideline. The rationale (nuxt 4.5 captures `$fetch` at import time, hence the call-time delegation) is documented in the PR description.

**Comment by DysektAI** [comment]
Addressed in f38e324a — the delegator now only forwards to per-test stubs that are actual vitest mocks (`vi.isMockFunction(current)`); the real nuxt `$fetch` (which `vi.unstubAllGlobals()` can restore to the global) is never invoked, so unstubbed requests always hit the fail-fast URL router and test isolation is preserved. Verified: full suite 2221/2221 green locally.

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "@commitlint/config-conventional": "^21.1.0",
     "@google/design.md": "^0.3.0",
     "@nuxt/eslint": "^1.15.2",
-    "@nuxt/test-utils": "^4.0.2",
+    "@nuxt/test-utils": "^4.1.0",
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply package.json:42]: Fixed in 20a1099a — removed the stale `@nuxt/test-utils>h3-next` override from `pnpm-workspace.yaml` (and the lockfile entry). Verified: `@nuxt/test-utils@4.1.0` now resolves with `h3: ^1.15.11` as its direct dependency; `h3-next` remains only as an optional peer (`>=2.0.1-rc.22`), which nothing requires at install time. The only remaining `h3-next: h3@2.0.1-rc.20` reference is an orphaned `@nuxt/test-utils@4.0.3` lockfile snapshot that nothing imports — harmless, pruned on next full re-resolution. Full suite 2221/2221 green.

**Comment by DysektAI** [comment]
## Production-readiness reconciliation

Reviewed the complete `origin/main...20a1099a` diff and all inline, review-summary, and top-level feedback.

- **CodeRabbit comment:** addressed in `d0ca4474`; the added `$fetch` rationale comments were removed per repository policy.
- **Cubic / CodeAnt `$fetch` isolation finding:** addressed in `f38e324a` and hardened in `20a1099a`; the delegator only forwards to active Vitest mocks, while restored real globals fall back to the fail-fast URL router. The full 2,221-test suite passes with the affected `unstubAllGlobals()` test files.
- **Cubic stale `h3-next` override:** addressed in `20a1099a`; the workspace override was removed. `@nuxt/test-utils@4.1.0` resolves its supported direct `h3@1.15.11` dependency, and the frozen lockfile installs cleanly.
- **Greptile vendor-core regex note:** no change required. The new regex has the same prefix-matching behavior as the old `includes()` checks, including `vue-router`, `vue-i18n`, and `pinia-plugin-persistedstate`. The Vite 8 production build succeeds and emits the dedicated Leaflet chunk at 148.8 kB.
- **Socket low-adoption warnings:** reviewed and accepted. `@vitejs/devtools-kit` and `unrouting` are transitive dependencies of the upgraded official Nuxt / Nuxt i18n packages, not new direct dependencies. Socket, Snyk, CodeQL, and CodeAnt SCA all pass with no vulnerability finding.
- **Latest CodeRabbit run:** rate-limited. Per repository policy, I performed a direct self-review of all six changed files and the resolved dependency graph; no P0–P3 production blocker or unaddressed finding remains.

Local readiness validation on Node 24.12.0 / pnpm 11.14.0:

- `pnpm install --frozen-lockfile --offline` — pass
- `pnpm run typecheck` — pass
- `pnpm run lint` — pass
- `pnpm run test` — 227 files / 2,221 tests pass
- `pnpm run i18n:check` — pass; existing Crowdin fallback differences remain informational
- `pnpm run validate:openapi` — pass
- `CI=true pnpm run build` — pass on Nuxt 4.5.1 / Vite 8.2.0 / Rolldown

All three review threads are resolved, the PR branch is clean, the Cloudflare preview deployed successfully, and every required CI/security check is passing. **Verdict: READY to merge.**
